swan-cern/files/swan_computing_config.py

```diff
@@ -137,7 +137,7 @@ async def _init_hadoop_secret(self):
         Create secret for Spark/Hadoop
         """
 
-        cluster = self.spawner.user_options[self.spawner.spark_cluster_field]
+        cluster = self.spawner.user_options.get(self.spawner.spark_cluster_field, 'none')
```

Comment:
I realize now that this change is not needed since `_init_hadoop_secret` is also protected by the `_spark_enabled` condition.

But what I would do is remove below:
```
 if cluster == 'none':
            return None
```

Reply:
Since the cluster can't be none here either, `_spark_enabled` already checked.

Reply:
As I said above, the change is not needed.